src/Planner/PlannerActionsVisitor.cpp

```diff
@@ -510,26 +510,31 @@ class PlannerActionsVisitorImpl
         void set(size_t level)
         {
             check(level);
-            mask |= uint64_t(1) << level;
+            if (level)
+                mask |= (uint64_t(1) << (level - 1));
         }
 
         void reset(size_t level)
         {
             check(level);
-            mask &= ~(uint64_t(1) << level);
+            if (level)
+                mask &= ~(uint64_t(1) << (level - 1));
         }
 
         void add(Levels levels) { mask |= levels.mask; }
 
-        size_t max() const { return 63 - __builtin_clzll(mask); }
+        size_t max() const
+        {
+            return mask ? (64 - __builtin_clzll(mask)) : 0;
```

Comment:
Now, as you changed the range level can be `0` (previously I think it was at least 1) I think we should change `__builtin_clzll` to `getLeadingZeroBits` as the return value is undefined in case of  `mask == 0`

Reply:
but I check that mask != 0 ... but ok, it looks simplier

Reply:
Ah sorry, you are right.